contracts/airswap/AirSwapFeeConnector.sol

```diff
@@ -0,0 +1,202 @@
+        feeVaultAddress = _newAddress;
+        require(feeVaultAddress != address(0), "invalid vault");
+        emit FeeVaultAddressChangedEvent(msg.sender, feeVaultAddress);
+    }
+
+    /// @notice Set the address of the AirSwap contract
+    /// @param _newAddress The new address
+    function setSwapERC20Address(address _newAddress) public onlyOwner {
+        swapERC20Address = _newAddress;
+        require(swapERC20Address != address(0), "invalid swapper");
+        emit SwapERC20AddressChangedEvent(msg.sender, swapERC20Address);
+    }
+
+    function calculateInputFee(uint256 _sendAmount) public view returns (uint256) {
+        return _sendAmount.mul(inputFeeInPoints).div(POINTS);
+    }
+
+    function calculateOutputFee(uint256 _receiveAmount) public view returns (uint256) {
+        return _receiveAmount.mul(outputFeeInPoints).div(POINTS);
+    }
+
+    /// @notice Swap one token for another.
+    /// @param _sender Address which is sending the tokens
+    /// @param _recipient Address to send the resulting tokens after collecting the output fee
+    /// @param _nonce A one time nonce
+    /// @param _expiry Date at which the original proposal will expire
+    /// @param _signerWallet Address of the market maker wallet
+    /// @param _signerToken Address of the token to convert to
+    /// @param _signerAmount Amount of resulting token from the conversion
+    /// @param _totalSenderAmount The amount to be sent before the input fee is collected.
+    /// @param _v v part of the ECDSA signature
+    /// @param _r r part of the ECDSA signature
+    /// @param _s s part of the ECDSA signature
+    function swap(
+        address _sender,
+        address _recipient,
+        uint256 _nonce,
+        uint256 _expiry,
+        address _signerWallet,
+        address _signerToken,
+        uint256 _signerAmount,
+        address _senderToken,
+        uint256 _totalSenderAmount,
+        uint8 _v,
+        bytes32 _r,
+        bytes32 _s
+    ) public {
+        require(feeVaultAddress != address(0), "invalid vault");
+        require(swapERC20Address != address(0), "invalid swapper");
+
+        SwapRequest memory swapRequest =
+            SwapRequest(
+                _sender,
+                _recipient,
+                _nonce,
+                _expiry,
+                _signerWallet,
+                _signerToken,
+                _signerAmount,
+                _senderToken,
+                _totalSenderAmount,
+                _v,
+                _r,
+                _s
+            );
+
+        // first we move all the funds here
+        require(
+            IERC20_(swapRequest.senderToken).transferFrom(
+                swapRequest.sender,
+                address(this),
+                swapRequest.totalSenderAmount
+            ),
+            "transfer failed 1"
+        );
+
+        // then we collect the input fee
+        uint256 inputFee = calculateInputFee(swapRequest.totalSenderAmount);
+        require(
```

Comment:
The transfer here can be combined with the output fee.
So will save some gas here